### PR TITLE
[WIP]Add auto-build action

### PR DESCRIPTION
**Pull request by Stevent-fei**

### CODE REVIEW

File: .github/workflows/auto-build.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Auto build image
+on:
+  issue_comment:
+    types:
+      - created
+jobs:
+  issue_comment:
+    name: Auto build image
+    if: ${{ github.event.comment.body == '/make build v1.20.14 ADM64' }}
```
Comment (bxy4543) [inline .github/workflows/auto-build.yml:9]: 'v1.20.14' and 'ADM64' are variables that need to be specified by passing in parameters.

Comment (Stevent-fei) [reply .github/workflows/auto-build.yml:9]: yes. I'll run through the logic first, and then add variables

**Comment by fanux** [comment]
You can test it in your own repository first.

File: .github/workflows/auto-build.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          tar -xvf kubernetes-server-linux-amd64.tar.gz
+          sudo cp ./kubernetes/server/bin/kubectl ./rootfs/bin/
+          sudo cp ./kubernetes/server/bin/kubeadm ./rootfs/bin/
+          sudo cp ./kubernetes/server/bin/kubelet ./rootfs/bin/
+          wget http://sealer.oss-cn-beijing.aliyuncs.com/auto-build/sealer.tar.gz
+          sudo tar -xvf sealer.tar.gz -C /usr/bin
+          sudo sealer login ${{secrets.PRIVATEWAREHOUSE}}
+          cd ./rootfs
+          cat>imageList<<EOF
+          k8s.gcr.io/kube-apiserver:v1.20.14
```
Comment (fanux) [inline .github/workflows/auto-build.yml:29]: Using kubeadm image list to got the imageList

Comment (cuisongliu) [reply .github/workflows/auto-build.yml:29]:
```
kubeadm config images list --config kubeadm.yaml
```

need add --config  param to fixed kubernetes version .default fetch images version is latest version in 1.20.x

**Comment by Stevent-fei** [comment]
The remaining trigger conditions need to be modified

File: .github/workflows/auto-build.yml (hunk 2 above)
Comment (Stevent-fei) [reply .github/workflows/auto-build.yml]: The premise of using kubedm is to install kubedm. I think setting it to variable will make all versions render, and I have tested it

File: .github/workflows/auto-build.yml [hunk 3]
```diff
@@ -0,0 +1,249 @@
+name: Auto build image
+on:
+  issue_comment:
+    types:
+      - created
+jobs:
+  issue_comment:
+    name: Auto build image
+    if: ${{ github.event.comment.body == '/buildv1.20.14' }}
```
Comment (bxy4543) [inline .github/workflows/auto-build.yml:9]: https://docs.github.com/en/actions/learn-github-actions/expressions#contains

use the contains function to trigger workflow

Comment (bxy4543) [reply .github/workflows/auto-build.yml:9]: The command should look like this：
/imagebuild version arch：/imagebuild v1.20.14 arm64

File: .github/workflows/auto-build.yml [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+    steps:
+
+      - name: Check outputs
+        run: |
+          BUILD=${{github.event.comment.body}}
+          echo ${BUILD#*d}
```
Comment (bxy4543) [inline .github/workflows/auto-build.yml:18]: image
you can split the string like this to get the version.

File: .github/workflows/auto-build.yml (hunk 3 above)
Comment (bxy4543) [reply .github/workflows/auto-build.yml:9]: > https://docs.github.com/en/actions/learn-github-actions/expressions#contains
> 
> use the contains function to trigger workflow

like this: startswith(github.event.comment.body, '/imagebuild')

File: .github/workflows/auto-build.yml [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+          EOF
+          cd ..
+          sudo sealer build -f Kubefile -m lite -t kubernetes:${BUILD#*d} .
+          sudo sealer push kubernetes:${BUILD#*d}
+      - name: Auto build image
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          issue-number: ${{ github.event.issue.number }}
+          body: |
+            kubernetes:v1.20.14
```
Comment (bxy4543) [inline .github/workflows/auto-build.yml:249]: What does the body do

File: .github/workflows/auto-build.yml [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+          wget https://dl.k8s.io/${BUILD#*d}/kubernetes-server-linux-amd64.tar.gz
+          tar -xvf kubernetes-server-linux-amd64.tar.gz
+          sudo cp ./kubernetes/server/bin/kubectl ./rootfs/bin/
+          sudo cp ./kubernetes/server/bin/kubeadm ./rootfs/bin/
+          sudo cp ./kubernetes/server/bin/kubelet ./rootfs/bin/
+          wget http://sealer.oss-cn-beijing.aliyuncs.com/auto-build/sealer.tar.gz
+          sudo tar -xvf sealer.tar.gz -C /usr/bin
+          sudo sealer login ${{secrets.PRIVATEWAREHOUSE}}
+          cd ./rootfs
+          cat>imageList<<EOF
```
Comment (bxy4543) [inline .github/workflows/auto-build.yml:30]: image

File: .github/workflows/auto-build.yml [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+    steps:
+
+      - name: Check outputs
+        run: |
+          commentbody="${{github.event.comment.body}}"
+          version=`echo "$commentbody"|cut -d" " -f2`
+          echo $version
+          arch=`echo "$commentbody"|cut -d" " -f3`
+          echo $arch
+          wget http://sealer.oss-cn-beijing.aliyuncs.com/auto-build/k8sv1.20.14.tar.gz
```
Comment (fanux) [inline .github/workflows/auto-build.yml:22]: I think you can from a BaseImage witch not contains kubernetes bin files, only has docker.

File: .github/workflows/auto-build.yml (hunk 6 above)
Comment (fanux) [reply .github/workflows/auto-build.yml:33]: Yes  you should use kubeadm to list all images.

File: .github/workflows/auto-build.yml [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+            criSocket: /var/run/dockershim.sock
+            taints: null
+
+          ---
+          apiServer:
+            certSANs:
+            - 127.0.0.1
+            - apiserver.cluster.local
+            - 10.103.97.2
+            - 172.16.0.65
```
Comment (fanux) [inline .github/workflows/auto-build.yml:59]: Those ips for what?

File: .github/workflows/auto-build.yml [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+          k8s.gcr.io/kube-apiserver:${BUILD#*d}
+          k8s.gcr.io/kube-controller-manager:${BUILD#*d}
+          k8s.gcr.io/kube-scheduler:${BUILD#*d}
+          k8s.gcr.io/kube-proxy:${BUILD#*d}
+          k8s.gcr.io/pause:3.2
+          k8s.gcr.io/etcd:3.4.13-0
+          k8s.gcr.io/coredns:1.7.0
+          EOF
+          cd ./etc/
+          cat>kubeadm.yml<<EOF
```
[the remark this answers is not in the document]
Comment (Stevent-fei) [reply .github/workflows/auto-build.yml:36]: If this is done, the environment variable will be invalidated

File: .github/workflows/auto-build.yml (hunk 7 above)
Comment (Stevent-fei) [reply .github/workflows/auto-build.yml]: Yes, this file does not contain kubernetes binaries

**Review by fanux — APPROVED** [review]

**Pull request merged**